### ISSUE TITLE
Querrying todos within a context via REST api does not seem to work properly.

### ISSUE THREAD

**Issue by MFreeze** [TracksApp/tracks#1931]
When I try to fetch the todos with a context by typing:

```
curl -u username:p4ssw0rd -H "Content-Type: text/xml" https://tracks.duvillie.eu/contexts/1/todos.xml
```

I get all my todos no matter what ID I provide.

When I look at the log, I get the following lines, that seems to indicate that the id request has been properly transfered by nginx and identified by tracks:

```
I, [2015-10-08T11:44:19.434571 #30988]  INFO -- : Started GET "/contexts/1/todos.xml" for 162.38.28.49 at 2015-10-08 11:44:19 +0200
I, [2015-10-08T11:44:19.444114 #30988]  INFO -- : Processing by TodosController#index as XML
I, [2015-10-08T11:44:19.444463 #30988]  INFO -- :   Parameters: {"context_id"=>"1"}
I, [2015-10-08T11:44:26.157837 #30988]  INFO -- : Completed 200 OK in 6713ms (Views: 0.8ms | ActiveRecord: 144.2ms)
```

But the returned xml contains all the tasks I've entered, including tasks belonging to another context.

If I try to fetch todos within an empty context, I get a RecordNotFound error:

```
I, [2015-10-08T11:38:06.470350 #30988]  INFO -- : Started GET "/contexts/4/todos.xml" for 162.38.28.49 at 2015-10-08 11:38:06 +0200
I, [2015-10-08T11:38:06.475861 #30988]  INFO -- : Processing by TodosController#index as XML
I, [2015-10-08T11:38:06.476202 #30988]  INFO -- :   Parameters: {"context_id"=>"4"}
I, [2015-10-08T11:38:06.609700 #30988]  INFO -- : Completed 404 Not Found in 133ms (ActiveRecord: 7.5ms)
F, [2015-10-08T11:38:06.621227 #30988] FATAL -- :
ActiveRecord::RecordNotFound (Couldn't find Context with 'id'=4 [WHERE `contexts`.`user_id` = ?]):
   app/controllers/todos_controller.rb:1336:in `get_not_done_todos'
   app/controllers/todos_controller.rb:17:in `index'
```

And I get the same error when I try to fetch todos within an unexisting context:

```
I, [2015-10-08T11:52:42.839854 #30988]  INFO -- : Started GET "/contexts/51/todos.xml" for 162.38.28.49 at 2015-10-08 11:52:42 +0200
I, [2015-10-08T11:52:42.845030 #30988]  INFO -- : Processing by TodosController#index as XML
I, [2015-10-08T11:52:42.845507 #30988]  INFO -- :   Parameters: {"context_id"=>"51"}
I, [2015-10-08T11:52:42.975034 #30988]  INFO -- : Completed 404 Not Found in 129ms (ActiveRecord: 6.5ms)
F, [2015-10-08T11:52:42.986357 #30988] FATAL -- :
ActiveRecord::RecordNotFound (Couldn't find Context with 'id'=51 [WHERE `contexts`.`user_id` = ?]):
  app/controllers/todos_controller.rb:1336:in `get_not_done_todos'
  app/controllers/todos_controller.rb:17:in `index'
```

I don't know if the following informations are relevant, but: 
- my install is a fresh install of tracks v2.30
- I use mysql backend
- I run ruby2.1

Thank you in advance.

**Comment by C-Otto** [comment]
Duplicate of #1726 I guess, please have a look at #2016 for a possible fix.

**Comment by RedNifre** [comment]
I interpreted the fix in #2016 to mean this, is this correct?:

contexts/ID/todos.xml?limit_to_active_todos=1

This doesn't work, I only get this:
<?xml version="1.0" encoding="UTF-8"?>
<nil-classes type="array"/>

Is there a work-around for this? Unlike projects/ID.xml, contexts/ID.xml doesn't contain the todos of that context...

**Comment by C-Otto** [comment]
Let's keep this discussion to one issue.

Did you apply the fix of #2016? In other words, did you change the code to make use of `@not_done_todos`? This variable, despite the name, should contain the filtered actions, in your case only the active actions.

**Comment by RedNifre** [comment]
I don't know that much about ruby, here's what I did:
- Shut down the app
- Edit the three files from the commit to change some @todos to @not_done_todos
- Restarted
- Used curl again to fetch i.e. curl -u bla:bla server/contexts/2/todos.xml
Is this the correct way to fix this?

I see no change, there are still todos from other contexts/projects and todos with the state "completed".

I don't need the API urgently, as a work-around I can filter client-side for now and wait for an official fix. Any idea when your fix will be released officially?

**Comment by C-Otto** [comment]
That should work. What is the response you get? Do the logs show anything helpful? Maybe you need to re-generate assets?

**Comment by RedNifre** [comment]
The response stayed the same, but maybe I made a misstake somewhere (Ruby newbie here).
Instead of trying to fix it I'll filter it client side and just wait for the next release.

Thank you for your help.

**Comment by ZeiP** [comment]
Seems like a duplicate of #2032, which was fixed in PR #2210.
